helpers/localize.js

```diff
@@ -0,0 +1,34 @@
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+export const allowedTags = Object.freeze(['d2l-link', 'd2l-tooltip-help', 'p', 'br', 'b', 'strong', 'i', 'em']);
+
+const markupError = `localizeHTML() rich-text replacements must use localizeMarkup templates with only the following allowed elements: ${allowedTags}. For more information, see: https://github.com/BrightspaceUI/core/blob/main/mixins/localize-mixin.md`;
+const validTerminators = '([>\\s/]|$)';
+const allowedAfterTriangleBracket = `/?(${allowedTags.join('|')})?${validTerminators}`;
+const disallowedTagsRegex = new RegExp(`<(?!${allowedAfterTriangleBracket})`);
+
+export function validateMarkup(content, applyRegex) {
+	if (content) {
+		if (content.map) return content.forEach(item => validateMarkup(item));
+		if (content._localizeMarkup) return;
+		if (Object.hasOwn(content, '_$litType$')) throw markupError;
+		if (applyRegex && content.constructor === String && disallowedTagsRegex.test(content)) throw markupError;
+	}
+}
+
+export function localizeMarkup(strings, ...expressions) {
+	strings.forEach(str => validateMarkup(str, true));
+	expressions.forEach(exp => validateMarkup(exp, true));
+	return { ...html(strings, ...expressions), _localizeMarkup: true };
+}
+
+export function generateLink({ href, target }) {
+	import('../components/link/link.js');
+	return chunks => localizeMarkup`<d2l-link href="${ifDefined(href)}" target="${ifDefined(target)}">${chunks}</d2l-link>`;
+}
+
+export function generateTooltipHelp({ contents }) {
+	import('../components/tooltip/tooltip-help.js');
+	return chunks => localizeMarkup`<d2l-tooltip-help inherit-font-style text="${ifDefined(chunks)}">${contents}</d2l-tooltip-help>`;
```

Comment:
Are we sure we want to invert the signature of tooltip-help like this? A couple thoughts:

- I don't like breaking the pattern of mirroring the component, with chunks being the contents and any attributes being passed into the function
- This could end up pretty broken - chunks can be anything, not just a string, while the `text` attribute needs a string only

Reply:
I'm not sure what you mean. This is how the component API works. I get that it's weird in an inline context like this, but without changing the component I don't see what can be done here.

It can only be as broken as when a dev adds the same stuff to `text` without doing so through `localizeHTML`. It'll be pretty obvious that it's broken before it goes off for translation.

Reply:
Oh, yeah it just can't work this way because of translations. We would be able to construct a regex that would split out the `chunks` into a whole separate term to translate (that would get put back together automatically), but the `text` is part of the original langterm so there's no way to get that from code into the translation file to go out as one whole langterm.

Reply:
Also, we always want `inherit-font-style`, no? If it's not meant to be inline then it doesn't need to be in the langterm.

Reply:
Okay, sounds reasonable 👍. I think we may be good to go then, except for the readme!

Reply:
I can take a swing at the README.